hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -896,17 +919,22 @@ private void finishRestore(final Timer.Context context, Map<String, List<HoodieR
   }
 
   private void rollbackInternal(String commitToRollback) {
-    if (commitToRollback.isEmpty()) {
-      logger.info("List of commits to rollback is empty");
-      return;
-    }
+
     final String startRollbackTime = startInstant();
     final Timer.Context context = startContext();
     // Create a Hoodie table which encapsulated the commits and files visible
     try {
-      List<HoodieRollbackStat> stats = doRollbackAndGetStats(commitToRollback);
-      Map<String, List<HoodieRollbackStat>> statToCommit = new HashMap<>();
-      finishRollback(context, stats, Arrays.asList(commitToRollback), startRollbackTime);
+      // Create a Hoodie table which encapsulated the commits and files visible
+      HoodieTable<T> table = HoodieTable.getHoodieTable(
+          createMetaClient(true), config, jsc);
+      List<HoodieInstant> rollbackInstants = table.getActiveTimeline().getCommitsTimeline().getInstants()
+          .filter(instant -> HoodieActiveTimeline.EQUAL.test(instant.getTimestamp(), commitToRollback))
+          .collect(Collectors.toList());
+      if (!rollbackInstants.isEmpty()) {
+        Preconditions.checkArgument(rollbackInstants.size() == 1);
+        List<HoodieRollbackStat> stats = doRollbackAndGetStats(rollbackInstants.get(0));
```

Comment:
Instead of checking and performing preconditions just do table.getActiveTimeline().getCommitsTimeline().getInstants()
          .filter(instant -> HoodieActiveTimeline.EQUAL.test(instant.getTimestamp(), commitToRollback)).findFirst.get() ?